### PR TITLE
[Tables] updating docstring

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py [hunk 1]
```diff
@@ -35,17 +35,20 @@
...
     def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
         self,
         endpoint: str,
         *,
-        credential: Optional[Union[AzureSasCredential, AzureNamedKeyCredential]] = None,
+        credential: Optional[Union[AzureSasCredential, AzureNamedKeyCredential, "AsyncTokenCredential"]] = None,
         **kwargs: Any
     ) -> None:
-        super(AsyncTablesBaseClient, self).__init__(endpoint, credential=credential, **kwargs)
+        super(AsyncTablesBaseClient, self).__init__(endpoint, credential=credential, **kwargs)  # type: ignore
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py:51]: Is this a problem that we are passing an async credential policy into the parent base_client constructor?

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py [hunk 2]
```diff
@@ -326,7 +326,7 @@ def get_table_client(self, table_name: str, **kwargs) -> TableClient:
         return TableClient(
             self.url,
             table_name=table_name,
-            credential=self.credential,
+            credential=self.credential,  # type: ignore
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py:329]: Why do we have to suppress the type here?

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py:51]: Yes, this gets passed all the way down to `AccountHostsMixin` which expects a sync token credential.

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py (hunk 2 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py:329]: The AccountHostsMixin expects a sync token credential.

**Review by annatisch — APPROVED** [review]

**Pull request merged**